packages/react-dom-bindings/src/shared/ReactDOMResources.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {ReactNodeList} from 'shared/ReactTypes';
+
+import isArray from 'shared/isArray';
+import {
+  REACT_ELEMENT_TYPE,
+  REACT_FORWARD_REF_TYPE,
+  REACT_FRAGMENT_TYPE,
+} from 'shared/ReactSymbols';
+
+export function concatTextChildrenProd(children: ReactNodeList): string {
+  if (__DEV__) {
+    throw new Error(
+      'concatTextChildrenProd should never be called when running in development mode. This is a bug in React.',
+    );
+  }
+  const combined = concatTextChildrenImpl(children);
+  invalidChild = null;
+  return combined;
+}
+
+export function concatTextChildrenDev(
+  children: ReactNodeList,
+  onInvalidChild: ReactNodeList => void,
```

Comment:
We haven't needed to use this pattern anywhere else. I'm curious why now.

Why not just call console.error in here?

Reply:
Yeah, at first I was thinking the error messages may need to be different, depending on Fizz/Client but it ended up making sense for them to be identical. I'll remove